### PR TITLE
add discussed options for scripting support

### PR DESCRIPTION
**Pull request by jreidinger**
## Problem

There is no documentation how to add custom scripts to autoinstallation.

see https://github.com/openSUSE/agama/discussions/990

## Solution

Document it including planned stuff.

### CODE REVIEW

File: autoinstallation/README.md [hunk 1]
```diff
@@ -154,6 +154,131 @@ set -ex
...
+##### Partitioning Changes
+
+Sometimes there is a tricky partitioning that needs to be modified after partitioning and before installing RPMs, such as changing the fstab and mount an extra partition.
+
+
+```sh
+set -ex
+
+/usr/bin/agama config set software.product=Tumbleweed
+
```
Comment (imobachgs) [inline autoinstallation/README.md:211]: `TOOMANYEMPTYLINES` :-)

File: autoinstallation/README.md [hunk 2]
```diff
@@ -154,6 +154,131 @@ set -ex
...
+/usr/bin/agama install --until partitioning # install till the partitioning step
+
+# Place for specific changes to /dev
+
+/usr/bin/agama install # do the rest of the installation
+```
+
+#### After Deployment
+
+Note: not supported now ( params to install is not implemented yet )
```
Comment (imobachgs) [inline autoinstallation/README.md:223]:
```suggestion
Note: not supported now (params to install is not implemented yet).
```

File: autoinstallation/README.md [hunk 3]
```diff
@@ -154,6 +154,131 @@ set -ex
...
+
+systemctl --root /mnt enable velociraptor-client
+
+/usr/bin/agama install # do the rest of the installation - basically unmount and copy logs
+
+```
+
+##### Tuning the Kernel
+
+Another type of change is changes that are necessary for a successful reboot, such as some kernel tuning or adding some remote storage that need to be mounted during boot.
```
Comment (imobachgs) [inline autoinstallation/README.md:258]:
```suggestion
Another scenario is when you need to make some changes required for a successful reboot, such as some kernel tuning or adding some remote storage that needs to be mounted during boot.
```

File: autoinstallation/README.md [hunk 4]
```diff
@@ -154,6 +154,131 @@ set -ex
...
+
+# Do custom modification of /mnt including call to dracut
+
+/usr/bin/agama install # do the rest of the installation - basically unmount and copy logs
+
+```
+
+#### After Reboot
+
+Users usually do a lot of things with post install scripts in autoyast e.g. calling zypper to install additional software, modify configuration files or manipulate with systemd services. This is done after the first reboot.
```
Comment (imobachgs) [inline autoinstallation/README.md:279]:
```suggestion
Users usually do a lot of things with post installation scripts in AutoYaST e.g. calling zypper to install additional software, modify configuration files or manipulate with systemd services. This is done after the first reboot.
```

**Review by imobachgs — CHANGES_REQUESTED** [review]
In general it looks good, thanks! Just some suggestions. Perhaps `TOOMANYEMPTYLINES` :wink:

File: autoinstallation/README.md [hunk 5]
```diff
@@ -154,6 +154,131 @@ set -ex
...
+#### Changes Before Installation
+
+##### Hardware Activation
+
+In some cases it is necessary to activate tricky devices manually before starting the installation. An example is two network cards, one for the external network and one for the internal network.
+
+```sh
+set -ex
+
+/usr/bin/agama profile download ftp://my.server/tricky_hardware_setup.sh
```
[the remark this answers is not in the document]
Comment (jreidinger) [reply autoinstallation/README.md:170]: yeap, that is something I am commenting when we add it. As it is generic download command that support also additional url schemas

File: autoinstallation/README.md [hunk 6]
```diff
@@ -154,6 +154,131 @@ set -ex
 You might want to have a look to [Agama's default script for inspiration](./scripts/auto.sh). Such a
 script comes into action when you provide a profile.
 
+### Support for Custom Scripts
+
+The goal of this section is to document examples and use cases for additional scripting support in agama autoinstall.
```
[the remark this answers is not in the document]
Comment (jreidinger) [reply autoinstallation/README.md:159]: yeah, spelling is always tricky :)

**Review by imobachgs — APPROVED** [review]

**Pull request merged**